### DISCUSSION TITLE
How to simulate keyboard events in code-server

### DISCUSSION THREAD

**Discussion by hkdeman** [coder/code-server#3929] [Q&A]
Hi,

I was trying to simulate Keyboard events like "Ctrl" + "p" (to open files) using javascript and I was unabel to do so. Any ideas on how this can be achieved? Is there function that can be called from document model? or simulation of actual keybaord events?

I tried to retrigger the same event when I humanly clicked those keys (and capture the event) but I get "TypeError: d.hasAttribute is not a function". Is there way out?

I have been trying all the above from console log for now. Thank you for the help :)

**Comment by jsjoeio** [comment]
Are you building a VS Code extension? 

You should be able to contribute a keybinding. See these:
- https://code.visualstudio.com/api/references/vscode-api#commands
- https://code.visualstudio.com/api/references/when-clause-contexts#add-a-custom-when-clause-context

Otherwise, you should be able to listen to keyboard input but I'm not sure how 🤔

Let me know if that helps!
